src/ManagedServices/ManagedServices/Extensions/ManagedServicesExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+using Microsoft.Azure.Management.ManagedServices.Models;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices
```

Comment:
Note that this common library:  https://github.com/Azure/azure-powershell-common/blob/master/src/ResourceManager/Version2016_09_01/Utilities/Models/ResourceIdentifier.cs  allows you to split a resource by components

Reply:
I tried to use that one but it doesnt work when the resourceid doesnt contain resourcegroups